File: frontend/src/fakeBackend/services/LocationService.js
import { db } from "../db";
import { generateId } from "../util/generateID";

export const LocationService = {
  getLocation(locationId) {
    return db.locations.find(loc => loc.id === locationId);
  },

  getLocationsForClient(clientId) {
    return db.locations.filter(loc => loc.clientId === clientId);
  },

  getAllLocations() {
    return db.locations.map(location => {
      const client = db.clients.find(c => c.id === location.clientId);
      return {
        ...location,
        clientName: client?.name
      };
    });
  },

  addLocation({ clientId, name, region }) {
    const location = {
      id: generateId(),
      clientId,
      name,
      region
    };
    db.locations.push(location);
    return location;
  }
};

File: backend/src/main/java/com/emission_impossible/carbon_modelling_tool/service/LocationService.java
package com.emission_impossible.carbon_modelling_tool.service;

import com.emission_impossible.carbon_modelling_tool.model.Client;
import com.emission_impossible.carbon_modelling_tool.model.Location;
import com.emission_impossible.carbon_modelling_tool.repository.ClientRepository;
import com.emission_impossible.carbon_modelling_tool.repository.LocationRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class LocationService {

    private final LocationRepository locationRepository;
    private final ClientRepository clientRepository;

    public LocationService(LocationRepository locationRepository, ClientRepository clientRepository){
        this.locationRepository = locationRepository;
        this.clientRepository = clientRepository;
    }


    public Location getLocation(Long locationId){
        return this.locationRepository.findById(locationId)
                .orElseThrow(() -> new RuntimeException(String.format("Location with location id %d not found", locationId)));
    }

    public List<Location> getLocationsForClient(Long clientId){
        return this.locationRepository.findByClientId(clientId);
    }

    public List<Location> getAllLocations(){
        return this.locationRepository.findAll();
    }

    public Location addLocation(Long clientId, String name, String region){

        Client findClient = this.clientRepository.findById(clientId)
                .orElseThrow(() -> new RuntimeException(String.format("Client with client id %d not found", clientId)));

        Location newLocation = new Location(name, region, findClient);

        this.locationRepository.save(newLocation);

        return newLocation;
    }

}
